### DISCUSSION TITLE
Asymmetric pitch bend and more

### DISCUSSION THREAD

**Discussion by MockbaTheBorg** [ElectroTechnique/electrotechnique.github.io#68] [New developments]
Hi ElectroTechnique,

I would like to ask for a couple new "features":
1 - Can we have two settings for pitchbend? One for up and one for down?
     This would be cool to have for a monophonic soloing lead.

2 - When in unison mode, can we set the detune pot to control the amount of spread between the voices?
     This would allow for controlling the amount of spread and make it gentler in some cases.

Not sure if this has been asked before. 
Also, if you point me to the correct files I can go and give it a try implementing it myself. I see the code is very modular, so for the pitchbend would be just a matter of appending an extra value to the patches (so it doesn't break compatibility) and adding option to the menu I guess.

I will try also adding my own waveforms just for the fun of it. I wonder how complicated it would be to add sync. (gotta deal with the band limiting)

Nice synth. I am happy I got it.

**Comment by Electrotechnique** [comment]
Hello, yes you can fork [https://github.com/ElectroTechnique/TSynth-Teensy4.1](https://github.com/ElectroTechnique/TSynth-Teensy4.1)

What do you mean by spread in Unison? The detuning amount between each voice?

**Reply by MockbaTheBorg** [reply]
Yes ... today the "spread" seems constant. And when I turn the detune knob it just lowers the frequency of the whole Unison.
I think it would be more useful if, instead of detuning all voices together, it just reduced the "spread". So we could make a more or less "rich" unison as we please.

**Reply by Electrotechnique** [reply]
Ok, have a look at Detune.cpp. There is a 4x24 array that detunes the 24 voices based on how many (up to four) notes are pressed in unison.

**Comment by MockbaTheBorg** [comment]
Cool. I will check it out.
I will do some research on the pitchbend too. Though this could probably
break patch file compatibility.

**Reply by Electrotechnique** [reply]
Probably not, pitchbend is global and stored in EEPROM.
